src/core/unit-serialize.c

```diff
@@ -151,6 +152,12 @@ int unit_serialize(Unit *u, FILE *f, FDSet *fds, bool serialize_jobs) {
         (void) serialize_cgroup_mask(f, "cgroup-enabled-mask", u->cgroup_enabled_mask);
         (void) serialize_cgroup_mask(f, "cgroup-invalidated-mask", u->cgroup_invalidated_mask);
 
+        if (u->ipv4_socket_bind_bpf_link)
+                (void) serialize_bpf_link(f, fds, "ipv4-socket-bind-bpf-link", u->ipv4_socket_bind_bpf_link);
+
+        if (u->ipv6_socket_bind_bpf_link)
+                (void) serialize_bpf_link(f, fds, "ipv6-socket-bind-bpf-link", u->ipv6_socket_bind_bpf_link);
```

Comment:
Maybe make the `!= NULL` check internally? It'd simplify this long function a bit.